File: Javascript/75/283.js
// 283. Move Zeroes
var moveZeroes = function (nums) {

  if (nums.length < 0) return 0
  
  let index = 0;
  for (let i = 0; i < nums.length; i++) {
    const el = nums[i]
    if (el != 0) {
      nums[index] = el;
      index++
    }
    
  }
  for (let i = index; i < nums.length; i++) {
    nums[i] = 0;
    
  }

  return nums

};

console.log(moveZeroes([0, 1, 0, 3, 12]));

File: Leetcode/283.py
class Solution:
    def moveZero(self, nums: list[int]) -> None:
        p = 0
        for index, element in enumerate(nums):
            if element != 0:
                nums[p] = element
                p += 1
        for i in range(p, len(nums)):
            nums[i] = 0
        return nums


nums = [0, 1, 0, 3, 12]

r = Solution().moveZero(nums)
print(r)
